edb/server/dbview/dbview.pyx

```diff
@@ -148,6 +160,73 @@ cdef class Database:
     def tenant(self):
         return self._index._tenant
 
+    def stop(self):
+        if self._cache_worker_task:
+            self._cache_worker_task.cancel()
+            self._cache_worker_task = None
+
+    async def cache_worker(self):
+        while True:
+            try:
+                await self._cache_worker()
+            except Exception as ex:
+                debug.dump(ex)
+                metrics.background_errors.inc(
+                    1.0, self.tenant._instance_name, "cache_worker"
+                )
```

Comment:
I'm usually very nervous about `while True` that doesn't wait before the next attempt. I'd add `await sleep(0.1)` to the `except` clause. In case of some errors you might want to give the system time to recover, and not saturate it with BS work until it's done.